src/Components/Aspire.Microsoft.EntityFrameworkCore.Cosmos/AspireAzureEFCoreCosmosDBExtensions.cs

```diff
@@ -24,24 +20,19 @@ public static class AspireAzureEFCoreCosmosDBExtensions
     private const DynamicallyAccessedMemberTypes RequiredByEF = DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors | DynamicallyAccessedMemberTypes.PublicProperties;
 
     /// <summary>
-    /// Registers the given <see cref="DbContext" /> as a service in the services provided by the <paramref name="builder"/>.
     /// Configures the connection pooling, logging and telemetry for the <see cref="DbContext" />.
     /// </summary>
     /// <typeparam name="TContext">The <see cref="DbContext" /> that needs to be registered.</typeparam>
-    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
-    /// <param name="connectionName">A name used to retrieve the connection string from the ConnectionStrings configuration section.</param>
-    /// <param name="databaseName">The name of the database to use within the Azure Cosmos DB account.</param>
+    /// <param name="services">The service collection.</param>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from.</param>
     /// <param name="configureSettings">An optional delegate that can be used for customizing settings. It's invoked after the settings are read from the configuration.</param>
-    /// <param name="configureDbContextOptions">An optional delegate to configure the <see cref="DbContextOptions"/> for the context.</param>
-    /// <exception cref="ArgumentNullException">Thrown if mandatory <paramref name="builder"/> is null.</exception>
-    /// <exception cref="InvalidOperationException">Thrown when mandatory <see cref="EntityFrameworkCoreCosmosDBSettings.ConnectionString"/> is not provided.</exception>
-    public static void AddCosmosDbContext<[DynamicallyAccessedMembers(RequiredByEF)] TContext>(
-        this IHostApplicationBuilder builder,
-        string connectionName,
-        string databaseName,
-        Action<EntityFrameworkCoreCosmosDBSettings>? configureSettings = null,
-        Action<DbContextOptionsBuilder>? configureDbContextOptions = null) where TContext : DbContext
+    /// <exception cref="ArgumentNullException">Thrown if mandatory <paramref name="services"/> or <paramref name="builder"/> is null.</exception>
+    public static IServiceCollection EnrichCosmosDbEntityFrameworkCore<[DynamicallyAccessedMembers(RequiredByEF)] TContext>(
+        this IServiceCollection services,
+        IHostApplicationBuilder builder,
+        Action<EntityFrameworkCoreCosmosDBSettings>? configureSettings = null) where TContext : DbContext
```

Comment:
Passing both of these look strange. Should it be IConfiguration?

Reply:
Most `Enrich` methods need to call `builder.TryAddHealthCheck()` so configuration alone is not sufficient.

Reply:
I see. This API is super strange.

Reply:
Mainly due to this comment https://github.com/dotnet/aspire/pull/1706#discussion_r1468217964

In retrospect I think it was better with two distinct called and no chaining. (no extension method on `IServiceCollection` bu on `IHostApplicationBuilder` instead)